low-level/sensors/impl/generic/raid_integrity_data.py

```diff
@@ -166,22 +165,27 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
+                    self._alert_msg = RaidAlertMsgs.STATE_MSG.value
+                    self._send_json_msg(self.alert_type, device, self._alert_msg)
```

Comment:
Started sending fault alert on inability to check the RAID state also? This is not signifying actual RAID fault as i see it, but at _check_mismatch_count() we get to know that! Am i misunderstanding something ?

Reply:
The alert will be send only the first time we get the status as "failed" i.e(when idle state is not found).
After that it just retry the execution every hour but not send the alert until status turns to success. (idle state is found)
Then it send the FAULT_RESOLVED alert.

Reply:
same behavior with mismatch_cnt scenario

Reply:
so this _retry_execution() is a blocking call, until 'success' returned for the given function.

To me, its confusing. We could have easily taken care in same 'for' loop (line 160), by making use of some bool flags to decide whether to execute next steps in each iteration, like 'raid_status_check', 'mismatch_count_check and without  _retry_execution(). 

But right now i dont to push for that